RetailCoder.VBE/UI/Command/AddTestModuleCommand.cs

```diff
@@ -63,18 +69,30 @@ public AddTestModuleCommand(IVBE vbe, RubberduckParserState state, IGeneralConfi
         private string GetTestModule(IUnitTestSettings settings)
         {
             var assertClass = settings.AssertMode == AssertMode.StrictAssert ? string.Empty : "Permissive";
-            var moduleBinding = settings.BindingMode == BindingMode.EarlyBinding
-                ? string.Format(ModuleEarlyBinding, assertClass)
-                : ModuleLateBinding;
 
-            var formattedModuleTemplate = string.Format(TestModuleEmptyTemplate, FolderAnnotation, moduleBinding);
+            bool isEarlyBinding = settings.BindingMode == BindingMode.EarlyBinding;
+            var moduleBinding = isEarlyBinding
+                    ? string.Format(AssertEarlyBinding,assertClass)
+                    : AssertLateBinding;
+            var moduleFakes = string.Format(ModuleFakesProvider, isEarlyBinding
+                                            ? "Rubberduck.IFakesProvider"
+                                            : "Object");
+
+            var formattedModuleTemplate = string.Format(TestModuleEmptyTemplate, FolderAnnotation, moduleBinding, moduleFakes);
 
             if (settings.ModuleInit)
             {
-                var lateBindingString = string.Format(ModuleInitLateBinding,
-                    settings.AssertMode == AssertMode.StrictAssert ? string.Empty : "Permissive");
+                var assertLateBindingString = string.Format(ModuleInitLateBinding, assertClass);
+                var assertEarlyBindingString = "Set Assert = New Rubberduck.AssertClass";
+
+                const string rdFakes = "Rubberduck.FakesProvider";
+                var fakesEarlyBindingString = $"New {rdFakes}";
+                var fakesLateBindingString = $"CreateObject({rdFakes})";
 
-                formattedModuleTemplate += string.Format(_moduleInit, settings.BindingMode == BindingMode.EarlyBinding ? string.Empty : lateBindingString);
+                formattedModuleTemplate += string.Format(_moduleInit,
+                                                        isEarlyBinding ? assertEarlyBindingString : assertLateBindingString,
+                                                        string.Format("Set Fakes = {0}", isEarlyBinding ? fakesEarlyBindingString : fakesLateBindingString)
```

Comment:
Logic is getting more complicated, and as we add unit testing features this method is subject to more changes; let's up the abstraction level a bit and pull some of the conditionals out of here and into their own dedicated methods: one that builds the declarations section, another that builds the `ModuleInitialize` method, another that builds the rest of the module, and this one that just puts all the pieces together.